x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypeConverter.java

```diff
@@ -313,6 +298,54 @@ public static TemporalAmount parseTemporalAmount(Object val, DataType expectedTy
         throw new ParsingException(Source.EMPTY, errorMessage, val, expectedType);
     }
 
+    public static Literal parseTemporalAmount(Expression e) {
```

Comment:
After thinking about it more, I'm leaning toward not letting it return a `null` now, let's see if it makes sense, otherwise will rename it to `maybeParseTemporalAmount` :). I'll change it to return a `TemporalAmount`, and the caller is responsible for building the `Literal`.